### ISSUE TITLE
[ic-radio-button] HideLabel prop set to true means label is not read out

### ISSUE THREAD

**Issue by GCHQ-Developer-299** [mi6/ic-ui-kit#4653]
### Summary of the bug

Raised by internal customer.
> I have found a bug in the IcRadioGroup component... when "hideLabel" is set to true, the input label prop is not being set as the aria-label.

### 🪜 How to reproduce

@GCHQ-Developer-299 has been able to replicate the issue in our Storybooks comparing ⁠[Ic Text Field](https://mi6.github.io/ic-ui-kit/branches/develop/react/?path=/story/text-field--playground) and ⁠[Radio Button](https://mi6.github.io/ic-ui-kit/branches/develop/react/?path=/story/radio-button--playground).

1. Go to Storybook with screenreader enabled
2. Confirm that text field / radio group reads the label out
3. Set hideLabel to true
4. Notice that radio group no longer reads the label

### 🧐 Expected behaviour

For the label to be read aloud even when not visible

### 🚨 Urgency (low, medium or high)

High

**Comment by k4its1t** [comment]
I’d like to take this. I’ll keep the visible-label path unchanged, give the radiogroup an accessible name from the existing label when hideLabel is enabled, add focused unit coverage (including the dynamic prop path), and run the web-components test, lint, and formatting checks.

**Comment by k4its1t** [comment]
I see #4655 now covers this issue, so I’m stepping back and won’t open a duplicate PR. Thanks for picking it up.
